### PR TITLE
Add want_ssh_host to foreman INI file

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Add new option to INI file building for foreman script

Connect https://github.com/ansible/awx/issues/2916

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION
The INI file via my own hacked testing system, and dummy parameters.

```
[foreman]
base_source_var = value_of_var
ssl_verify = False
url = https://foo.invalid
user = fooo
password = fooo

[ansible]
group_patterns = foo_group_patterns
want_facts = True
want_hostcollections = True
group_prefix = foo_group_prefix
want_ansible_ssh_host = True

[cache]
path = /tmp
max_age = 0

```

See expected format at:

https://github.com/ansible/ansible/blob/devel/contrib/inventory/foreman.ini#L159

### CODE REVIEW

File: awx/main/tasks.py [hunk 1]
```diff
@@ -1910,6 +1910,7 @@ def build_private_data(self, inventory_update, **kwargs):
             group_patterns = '[]'
             group_prefix = 'foreman_'
             want_hostcollections = 'False'
+            want_ssh_host = 'False'
```
Comment (kdelee) [inline awx/main/tasks.py:1913]: This is default value?

File: awx/main/tasks.py [hunk 2]
```diff
@@ -1919,6 +1920,8 @@ def build_private_data(self, inventory_update, **kwargs):
                     group_prefix = v
                 elif k == 'satellite6_want_hostcollections' and isinstance(v, bool):
                     want_hostcollections = v
+                elif k == 'satellite6_want_ssh_ansible_host' and isinstance(v, bool):
```
Comment (kdelee) [inline awx/main/tasks.py]: oopsy! should be `satellite6_want_ansible_ssh_host` for our dear docs readers to be able to prognosticate what to put

Extra sharing moment:
---------------------------------
I read docs and it says (my interpretation) "You see a var you want to set in foreman.ini!? Cool, write it here except put `satellite6`, e.g. `satellite6_my_fav_var`" in front

There is a difference between `satellite6` and `satellite6_` !!!!
Also it is very hard to spell word.
also, what if I am using katello or foreman?
:crying_cat_face:

**Review by kdelee — CHANGES_REQUESTED** [review]
need to flip flop some words, all need to be `want_ansible_ssh_host`

File: awx/main/tasks.py [hunk 3]
```diff
@@ -1919,6 +1920,8 @@ def build_private_data(self, inventory_update, **kwargs):
                     group_prefix = v
                 elif k == 'satellite6_want_hostcollections' and isinstance(v, bool):
                     want_hostcollections = v
+                elif k == 'satellite6_want_ansible_ssh_host' and isinstance(v, bool):
```
Comment (kdelee) [inline awx/main/tasks.py:1923]: :1st_place_medal:

**Review by kdelee — APPROVED** [review]

File: awx/main/tasks.py (hunk 2 above)
Comment (AlanCoding) [reply awx/main/tasks.py]: oh my gosh, thanks so much for finding that

File: awx/main/tasks.py (hunk 1 above)
Comment (AlanCoding) [reply awx/main/tasks.py:1913]: It's given in the example ini file as hard-coded to False.

You have to check the actual code to find the true default behavior:

https://github.com/ansible/ansible/blob/devel/contrib/inventory/foreman.py#L121

It does appear to set it to False.

I would still prefer to not include options we want to obtain the defaults for, but that ship has sailed.

File: awx/main/tasks.py (hunk 3 above)
Comment (wenottingham) [reply awx/main/tasks.py:1923]: why would we do this rather than just support a direct `want_ansible_ssh_host` source var?

**Comment by ryanpetrello** [comment]
@AlanCoding @kdelee does this still need to be a separate PR, or do you want to fold this into the inventory plugin updates?

**Comment by AlanCoding** [comment]
We have already produced the config file's schema for testing the transition to inventory plugins, and thus, this parameter is not included in that. I would strongly prefer to not include this with the inventory plugins changes, or merge it before that lands, for the sake of testing complexity.

It might be a good case-in-point for adding new fields after that lands, refactoring it to the new system, which should hopefully be a bit easier.

**Pull request closed without merging**